cpp/roaring.hh

```diff
@@ -61,6 +61,15 @@ public:
         api::roaring_bitmap_add_many(&roaring, n, data);
     }
 
+    /**
+     * Construct a bitmap from an initializer list.
+     */
+    Roaring(std::initializer_list<uint32_t> l) : Roaring() {
+        for(uint32_t x : l) {
+            add(x);
+        }
+    }
+
```

Comment:
I had one more (long) comment here and I can't find now so you might see it twice.

Basically I am concerned about the interaction between initializer lists and uniform initialization syntax.

Consider a hypothetical future world where you have (say) a two-arg constructor and an initializer list and a customer who likes uniform initialization syntax.

Library:
```
Roaring(int a, int b) {...}
Roaring(std::initializer_list<uint32> l) {...}
```

client:
```
// Note curly braces. This is valid syntax for the two-arg constructor but the compiler will
// prefer the initializer_list constructor.
Roaring r{5, 100};  // calls initializer_list constructor
```

I see two valid approaches:
1. Don't worry too much about the vagaries of uniform initialization syntax
2. Take away the initializer_list constructor and force customers to write their code "factory-style", as in

```
Roaring r = Roaring::bitmapOfList({3,4,,5})
```

which by the way is just as efficient.

Reply:
Ok. I will pull off constructors from this PR.

Reply:
Do we ever foresee such a constructor with two integers?

I can't think of a case where I would think we'd ever want `Roaring x{5, 100}` to make anything _but_ a bitmap which contains only 5 and 100.

The common example of where such a constructor is problematic is the `std::vector<int> x{4, 2}` vs `std::vector<int>(4, 2)` (which create a vector containing 4 and 2, and 4 copies of 2 respectively). Honestly, I'd say if the vector api were being created now, the initializer list constructor would stay, and the fill constructor would go or be moved to a static method.

Reply:
Two args was just an example. The issue arises for constructors of any arity, so I think your best argument would have to assert that Roaring will never want a one-argument integer or pointer constructor either. But note, it actually *already has* conflicting one and two-arg constructors, to wit:

```
Roaring64Map x{0};   // this compiles today
```
The above currently invokes `Roaring64Map::Roaring64Map(roaring_bitmap_t *s)` with `nullptr`. WIth an initializer_list constructor the meaning would change, and the initializer_list constructor would be preferred.

And also
```
Roaring64Map a{0, 0};  // almost compiles today
```
and the only reason it *doesn't* compile is because there are *two* constructors to bind to and the compiler can't decide which one:

```
Roaring64Map::Roaring64Map(size_t n, const uint32_t *data)
Roaring64Map::Roaring64Map(size_t n, const uint64_t *data)
```

Yes these are nitpicky edge-case examples and calling those *particular* constructors with `nullptr` is not something anyone would want anyway. But this is different from saying they don't exist. My point is that it merits a discussion a little bigger than the scope of this PR.

TL;DR:
1. I assumed the scope of this particular PR was to provide an `initializer_list` alternative to the existing error-prone varargs `bitmapOf`. I think it has accomplished that and so this PR is a good quantum of change as is.
2. If my opinion was sought I'd say I had assumed `bitmapOf` was really only useful for unit tests, but if the community wanted the constructor I'd at least ask that such a change (along with an initializer_list `operator=` by the way) be put in its own PR so that the project owners can understand the pros and cons.

Reply:
In modern C++ can't these be handled by creating ctors with std::nullptr_t types?  Differentiating between ctors that take a pointer versus an integer WRT NULL I think is a solved problem...?  I understand if the hope is to defer this discussion until later.

Reply:
Ah, that's a good suggestion. Yes, I am just proposing we proceed with caution and with a discussion.